File: examples/joystick12-example.cxx
/*
 * Author: Alexander Komarov <alexander.komarov@intel.com>
 * Copyright (c) 2014 Intel Corporation.
 *
 * Permission is hereby granted, free of charge, to any person obtaining
 * a copy of this software and associated documentation files (the
 * "Software"), to deal in the Software without restriction, including
 * without limitation the rights to use, copy, modify, merge, publish,
 * distribute, sublicense, and/or sell copies of the Software, and to
 * permit persons to whom the Software is furnished to do so, subject to
 * the following conditions:
 *
 * The above copyright notice and this permission notice shall be
 * included in all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
 * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
 * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
 * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
 * LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
 * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
 * WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

#include <unistd.h>
#include <iostream>
#include <signal.h>
#include "joystick12.h"
#include <stdlib.h>
#include <sys/time.h>

int is_running = 0;
upm::Joystick12 *sensor = NULL;

void
sig_handler(int signo)
{
    printf("got signal\n");
    if (signo == SIGINT) {
        is_running = 1;
    }
}

//! [Interesting]
int
main(int argc, char **argv)
{
    // Instantiate a joystick on analog pins A0 and A1
    sensor = new upm::Joystick12(0,1);
    signal(SIGINT, sig_handler);

    // Print the X and Y input values every second
    while (!is_running) {
        float x = sensor->getXInput();
        float y = sensor->getYInput();
        std::cout << "Driving X:" << x << ": and Y:" << y << std::endl;
        sleep(1);
    }

    std::cout << "exiting application" << std::endl;

    delete sensor;

    return 0;
}
//! [Interesting]

File: examples/java/Joystick12_Example.java
/*
 * Author: Stefan Andritoiu <stefan.andritoiu@intel.com>
 * Copyright (c) 2015 Intel Corporation.
 *
 * This program and the accompanying materials are made available under the
 * terms of the The MIT License which is available at
 * https://opensource.org/licenses/MIT.
 *
 * SPDX-License-Identifier: MIT
 */

public class Joystick12_Example {

	public static void main(String[] args) throws InterruptedException {
		//! [Interesting]
		// Instantiate a joystick on analog pins A0 and A1
		upm_joystick12.Joystick12 joystick = new upm_joystick12.Joystick12(0,1);

		// Print the X and Y input values every second
		while(true){
			float x = joystick.getXInput();
			float y = joystick.getYInput();

			System.out.println("Driving X: " + x + " , and Y: " + y );

			Thread.sleep(1000);
		}
		//! [Interesting]
	}

}
